### DISCUSSION TITLE
FINN transform LSTM to FPGA RTL question.

### DISCUSSION THREAD

**Discussion by gfwangjg** [Xilinx/finn#1054] [Q&A] [answered]
Hello, I was using Brevitas build QuantLSTM model.
I success export quant_lstm.onnx
But when I import quant_lstm.onnx and transformation it to QONNX

The OP in the model are below:

Input tensor datatype: FLOAT32
Output tensor datatype: FLOAT32
List of node operator types in the graph: 
['Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'Quant', 'QuantLSTMCell', 'Unsqueeze', 'Unsqueeze']

The Qonnx give error:

Exception: Op QuantLSTMCell not found in custom opset qonnx.custom_op.general

so my questions is , how can I use FINN to transform LSTM to FPGA RTL.

Thx.

**Comment by fpjentzsch** [comment, marked as answer]
Hi,
FINN does not natively support LSTMs. This paper linked on the FINN website describes an extension to support LSTMs that (to my knowledge) was never fully integrated: https://arxiv.org/pdf/1807.04093

**Reply by gfwangjg** [reply]
Thx, I'll check this paper.

**Comment by gfwangjg** [comment]
FINN dose not natively support LSTMs.
